components/brave_wallet/browser/solana_tx_manager.cc

```diff
@@ -830,6 +838,198 @@ void SolanaTxManager::MakeTxDataFromBase64EncodedTransaction(
                           mojom::SolanaProviderError::kSuccess, "");
 }
 
+void SolanaTxManager::MakeBubbleGumProgramTransferTxData(
+    const std::string& chain_id,
+    const std::string& token_address,
+    const std::string& from_wallet_address,
+    const std::string& to_wallet_address,
+    MakeBubbleGumProgramTransferTxDataCallback callback) {
+  // Get asset and proof data from SimpleHash
+  auto internal_callback =
+      base::BindOnce(&SolanaTxManager::OnFetchCompressedNftProof,
+                     weak_ptr_factory_.GetWeakPtr(), from_wallet_address,
+                     to_wallet_address, std::move(callback));
+
+  simple_hash_client_->FetchSolCompressedNftProofData(
+      token_address, std::move(internal_callback));
```

Comment:
nit: Maybe better to have this exposed through json_rpc_service (which just calls into simple hash client) like other methods we added, then we don't need to have a separate simple hash client created here, wdyt?